dev/io.openliberty.microprofile.telemetry.internal.common/src/io/openliberty/microprofile/telemetry/internal/common/info/OpenTelemetryInfoFactoryImpl.java

```diff
@@ -185,23 +218,87 @@ private static boolean checkDisabled(Map<String, String> oTelConfigs) {
             return new HashMap<String, String>();
         }
     }
 
+    private HashMap<String, String> getServerTelemetryProperties() {
+        try {
+            HashMap<String, String> telemetryProperties = new HashMap<>();
+
+            AccessController.doPrivileged(new PrivilegedAction<String>() {
+                @Override
+                public String run() {
+                    Map<String, String> envProperties = System.getenv();
+                    Map<Object, Object> systemProperties = System.getProperties();
+
+                    HashMap<String, String> tempProperties = new HashMap<>();
+
+                    //Check system properties for all configured OTEL properties
+                    for (Object propertyName : systemProperties.keySet()) {
+                        String normalizedName = ((String) propertyName).toLowerCase().replace('_', '.');
+                        if (normalizedName.startsWith("otel")) {
+                            String propertyValue = (String) systemProperties.get(propertyName);
+
+                            if (propertyValue != null) {
+                                tempProperties.put(normalizedName, propertyValue);
+                            }
+                        }
+
+                    }
+
+                    //Check system environment for all configured OTEL properties and replace any
+                    //previously configured values found in the system properties.
+                    for (String propertyName : envProperties.keySet()) {
+                        if (propertyName.startsWith("otel") || propertyName.startsWith("OTEL")) {
+
+                            String normalizedName = propertyName.toLowerCase().replace('_', '.');
+                            String propertyValue = envProperties.get(propertyName);
+
+                            if (tempProperties.containsKey(normalizedName))
+                                tempProperties.remove(normalizedName);
+
+                            if (propertyValue != null)
+                                tempProperties.put(normalizedName, propertyValue);
+                        }
+                    }
+
+                    //Only add telemetry properties if OTEL is enabled.
+                    if (tempProperties.containsKey(OpenTelemetryConstants.CONFIG_DISABLE_PROPERTY)
+                        && "false".equalsIgnoreCase(tempProperties.get(OpenTelemetryConstants.CONFIG_DISABLE_PROPERTY))) {
+                        telemetryProperties.putAll(tempProperties);
+                    }
+                    return null;
+                }
+            });
+
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "Runtime OTEL instance is being configured with the properties: {0}", telemetryProperties);
+            }
+
+            return telemetryProperties;
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Normally, we don't use `printStackTrace()`

Reply:
Thank you, outdated debug code. Removed.